### PR TITLE
Rearrange type parameters around Block trait

### PR DESCRIPTION
**Pull request by zakarumych**

### CODE REVIEW

File: src/arena.rs [hunk 1]
```diff
@@ -63,7 +64,7 @@ impl<T> ArenaAllocator<T> {
     fn cleanup<B, A>(&mut self, owner: &mut A, device: &B::Device)
     where
         B: Backend,
-        T: Block<B>,
+        T: Block<Memory = B::Memory>,
         A: MemoryAllocator<B, Block = T>,
```
Comment (kvark) [inline src/arena.rs:68]: could do this instead of having `T`:
```rust
where
    A::Block: Block<Memory = B::Memory>
```

Comment (zakarumych) [reply src/arena.rs:68]: No.
`A::Block: Block<Memory = B::Memory>` is tautology for `A: MemoryAllocator<B>`.
While `T` is a type parameter of `ArenaAllocator` itself. And it must be the type of superblock (`A::Block`)

**Comment by kvark** [comment]
Thanks!
bors r+

**Comment by zakarumych** [comment]
I'll merge it. Hope we have bors here soon.

**Pull request merged**